/*
This proto file implement

Build command:
python3 -m grpc_tools.protoc -I. --python_out=. --pyi_out=. --grpc_python_out=. ./mafiaRPC.proto
*/

syntax = "proto3";

package mafia;

service MafiaClient {
  rpc GetNewPlayerId(Request) returns (PlayerId) {}
  rpc Subscribe(Player) returns (stream Response) {}
  rpc Unsubscribe(Player) returns (Response) {}
  rpc SendVote(VoteRequest) returns (VoteResponse) {}
}

enum Status {
  FAIL = 0;
  SUCCESS = 1;
  START_GAME = 2;
  YOU_KILLED = 3;
  DAY_VOTE = 4;
  NIGHT_VOTE = 5;
  ROLE_DISTRIBUTION = 6;
  END_GAME = 7;
  MORNING_NOTIFICATION = 8;
}

message Request {
  string message = 1;
}

message PlayerId {
  uint32 id = 1;
}

message Player {
  int32 id = 1;
  string name = 2;
  string address = 3;
}

message Response {
  optional Status status = 1;
  string data = 2;
}

message VoteRequest {
  int32 session_id = 1;
  int32 player_id = 2;
}

message VoteResponse {
  Status status = 1;
}
